### PR TITLE
add plaid.Date and plaid.DateTime to source

### PR DESCRIPTION
**Pull request by ntindall**
see https://github.com/plaid/plaid-go/pull/46 for the motivation

cc @icholy @cbelsole

### CODE REVIEW

**Comment by ntindall** [comment]
@michaelckelly or @pbernasconi - can you help debug the circle ci issues?

File: plaid/time.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+// DateFormat defines the date format used for client-facing date strings.
+const DateFormat = "2006-01-02"
+
+// DateTimeFormat defines the datetime format used for client-facing date
+// strings.
+const DateTimeFormat = time.RFC3339
+
+// Date is a time.Time object whose default string format (including for json
+// serialization) is a client-facing date string.
+type Date time.Time
```
Comment (icholy) [inline plaid/time.go]: The standard practice is to use embedding here.

``` go
type Date struct { time.Time }
```

Comment (cbelsole) [reply plaid/time.go]: @icholy can you provide a link to the standard?

Comment (icholy) [reply plaid/time.go]: What?

Comment (cbelsole) [reply plaid/time.go]: I was wondering if you could provide a link that says that `type Date struct { time.Time }` is the way it should be done.

Comment (ntindall) [reply plaid/time.go]: I don't agree that it should be embedded in an anonymous struct, that will prevent callers from utilizing the existing `time.Time` functions, unless I am misunderstanding what you mean exactly @icholy

Comment (icholy) [reply plaid/time.go]: @ntindall When you create a new type (the way you're doing) it prevents callers from using any methods on `time.Time` without explicitly casting.

**Comment by ntindall** [comment]
@jeeyoungkim @themichaellai - can you review this?

**Comment by pbernasconi** [comment]
Closing this as stale

**Pull request closed without merging**